mlx/backend/metal/device.cpp

```diff
@@ -242,15 +242,15 @@ Device& device(mlx::core::Device) {
   return metal_device_;
 }
 
-NS::AutoreleasePool*& thread_autorelease_pool() {
+NS::AutoreleasePool*& Device::g_thread_autorelease_pool() {
```

Comment:
`g_` stands for gpu?

Reply:
Lol @andresy doesn't like the name. It stands for Global (since it's one per thread)